RecoLocalCalo/HGCalRecProducers/interface/HGCalLayerTiles.h

```diff
@@ -0,0 +1,68 @@
+// Authors: Felice Pantaleo - felice.pantaleo@cern.ch
+// Date: 03/2019
+// Copyright CERN
+
+#ifndef RecoLocalCalo_HGCalRecAlgos_HGCalLayerTiles
+#define RecoLocalCalo_HGCalRecAlgos_HGCalLayerTiles
+
+#include "RecoLocalCalo/HGCalRecProducers/interface/HGCalTilesConstants.h"
+
+#include <vector>
+#include <array>
+#include <cmath>
+
+class HGCalLayerTiles {
+public:
+  HGCalLayerTiles() { tiles_.resize(hgcalTilesConstants::nColumns * hgcalTilesConstants::nRows); }
+
+  void fill(const std::vector<float>& x, const std::vector<float>& y) {
+    auto cellsSize = x.size();
+    for (unsigned int i = 0; i < cellsSize; ++i) {
+      tiles_[getGlobalBin(x[i], y[i])].push_back(i);
+    }
+  }
+
+  int getXBin(float x) const {
+    constexpr float xRange = hgcalTilesConstants::maxX - hgcalTilesConstants::minX;
+    static_assert(xRange >= 0.);
+    constexpr float r = hgcalTilesConstants::nColumns / xRange;
+    int xBin = (x - hgcalTilesConstants::minX) * r;
+    xBin = std::min(xBin, hgcalTilesConstants::nColumns);
```

Comment:
`xBin = std::clamp(xBin, 0, hgcalTilesConstants::nColumns);` (and similarly below)

Reply:
thanks, done